### ISSUE TITLE
Set up Netlify CMS for eick.com

### ISSUE THREAD

**Issue by budparr** [theNewDynamic/mfocus-tracking#4]
This is a Jekyll site so we can't use our pre-built module. 

- [x] Set up Netlify CMS fields
- [ ] Set up Auth
- [x] Test

Need to know:

- [x] Do we use Editorial Workflow? (previews) - answer: yes

I believe this site is not on Netlify, so we'll need auth information, so this issue is dependent upon #2 being completed.

Site: https://github.com/missionfocus/www.missionfocus.com

**Comment by andreweick** [comment]
This site is on netlify

**Comment by andreweick** [comment]
I'm not sure I understand what "Editorial Workflow" means --

**Comment by budparr** [comment]
Editorial workflow does a few things: When you create a new entry, the CMS creates a branch/PR for that entry. This allows a PR-like process from within the CMS, such as approval stages of an entry. That's helpful in a multi-author environment. However, it also kicks in Netlify's deploy preview functionality, which allows you to also see the entry live on the branch of your site. That's pretty handy.

It takes a bit of getting used to (editing an existing entry also creates a branch), but the preview is nice.

**Comment by andreweick** [comment]
Sounds good — the “multi author” doesn’t really apply, but the previews off branches with PR’s is good

**Comment by andreweick** [comment]
Could this issue be worked first?  After this issue is completed, I can test it out // add content and test posts.

**Comment by budparr** [comment]
Sure. We had intended to knock out the Jekyll errors first because we think it will be short, but can start with this one instead. We need the credentials here (for me, and my team). I'll send you an email with who we need them for.

**Comment by regisphilibert** [comment]
Hello @andreweick. Is `git-gateway` enabled on Netlify for eick.com? I received your invites, but the activation link does not seem to produce anything on my end. 

Have you been though the `/members` section of you Netlify team dashboard and added me through the `Add member` process?

**Comment by regisphilibert** [comment]
@andreweick 
Pending we resolve the Netlify Identity issue mentioned above, I setup a Netlify instance for this repo tracking the branch [`add-neltifycms`](https://github.com/imintel/eick.com/tree/add-netlifycms
)

Site: https://tnd-eick.netlify.app/
CMS Dashboard: https://tnd-eick.netlify.app/cms

You should be able to create an account on the `/cms` page.

This allow you to tryout the CMS and make edits without changing the production site while still being able to see them deploy on https://tnd-eick.netlify.app/

If you think of anything you need and we did not add, just let me know.

**Comment by andreweick** [comment]
@regisphilibert Sorry I totally missed the update -- I just tried to update my notification settings --  I've been excited for your work!

Ok, Yes, I just enabled git-gateway, it **was not** enabled before...Can you check the config and see if I did it correctly?

**Comment by regisphilibert** [comment]
> If you think of anything you need and we did not add, just let me know.

If you think it's all good, we'll merge the branch into master.

**Comment by andreweick** [comment]
Ok, I tried the /cms directory, I tried to login w/ my netlify identity, but (surely) it's not there.

I say throw caution to the wind and just send a PR to merge to master -- I mean what could go wrong :-)

(Although, I didn't study the TDN-netlify-cms plugin much yet -- how do you handle upgrades / mods to the CMS part?)

**Comment by andreweick** [comment]
@regisphilibert I tried to merge that branch into master there were some netlify errors -- https://github.com/imintel/eick.com/pull/43/commits/f030bfb9258d08230d074f5388630432cc5e6a56 -- I'm not totally familiar with netlify publishing but I did see some reference s to the new URLs you listed above -- SO weird, for the life of me I couldn't find the URLs in the code that would cause the error.

If you could just send a PR when it's ready to be merged into (and pointed to) www.eick.com

**Comment by regisphilibert** [comment]
@andreweick I reopened the PR. I think the URL thing is front end. But I did need to update the branch name (removes it so it defaults to master).

I'm not sure how you're your Netlify identity is setup. But if it auto-register is activated, could you enable GitHub as a signin/register service? This way we'll be able to sign up with our GitHub creds.

Thanks.

**Comment by regisphilibert** [comment]
@andreweick I'm looking at some build errors and will get back to you.

**Comment by regisphilibert** [comment]
@andreweick it's all good. You can test the CMS here: https://deploy-preview-43--tnd-eick.netlify.app/cms/

I managed to signin using the GitHub button at the bottom.

WARNING: Be aware any changes made on this branch from the CMS Dashboard will be committed to master and therefor deployed to production.

**Comment by andreweick** [comment]
@regisphilibert Thanks -- although I'm still a bit confused?

I sent @budparr an issue this weekend, I want to migrate the (source) repo to my personal GitHub (rather than my company) -- in your opinion, does it make sense for me to change ownership of the repo (to GitHub.com/andreweick) and then update all the corresponding netlify bits to work, or should I 'squash merge' this https://github.com/imintel/eick.com/pull/43 and then do the migration.

This site isn't that precious to me (yet!) so a few days of downtime / migration are no problem -- I don't need to 'validate' with the TND' Domain you set up.

I'd rather move the site to it's final REPO and URL and then figure out this 'netlify identity'.

**Comment by budparr** [comment]
I will be review your requests from the weekend sometime today

**Comment by andreweick** [comment]
@regisphilibert https://tnd-eick.netlify.app/cms/ looks great.  

Although I can't get logged in at the moment, it was working for me a few nights ago and looked great.

I wasn't sure how to invoke the 'editorial' workflow (that would post on non-master branches), but that might be user error?

I want to move the 'source' repo from the IMITNEL group to my personal group.  Do you recommend I just 'delete' my netlify setup, move the repo, and then 're-create' the netlify entry?

**Comment by regisphilibert** [comment]
> I wasn't sure how to invoke the 'editorial' workflow (that would post on non-master branches), but that might be user error?

I just opened a PR to add the setting on your master branc.

> I want to move the 'source' repo from the IMITNEL group to my personal group. Do you recommend I just 'delete' my netlify setup, move the repo, and then 're-create' the netlify entry?

I think you can just do the repo move and, and then from your Netlify Instance update the repository.
![Capture d’écran, le 2020-07-24 à 09 53 11](https://user-images.githubusercontent.com/1480503/88398570-91e7de00-cd93-11ea-8772-1024ef491551.png)

But @budparr might be more knowledgable in this field.

**Comment by andreweick** [comment]
@regisphilibert @budparr well that was painless! I moved the repo to https://github.com/andreweick/eick.com, updated netlify and, republished, and all seems good -- I'm moving over the "old style drafts: true" to "editorial" ...
